src/components/Person.jsx

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import './Person.scss';
+import PropTypes from 'prop-types';
+
+export const Person = ({ person }) => {
+  const {
+    name,
+    age,
+    sex,
+    isMarried,
+    partnerName,
+  } = person;
+
+  return (
+    <section className="Person">
+      <h2 className="Person__name">{`My name is ${name}`}</h2>
+      {age && (
+        <p className="Person__age">{`I am ${age}`}</p>
+      )}
+      {(isMarried && (
+        sex === 'f' ? (
+          <p className="Person__partner">{`My husband's name is ${partnerName}`}</p>
+        ) : (
+          <p className="Person__partner">{`My wife's name is ${partnerName}`}</p>
+        )
+      )) || 'I am not married'}
```

Comment:
Will be better to create separate variable before return statement in component, save here final married status message and use this variable in jsx
Such way helps to avoid adding some code logic in jsx for better understanding final html

Reply:
i hardly catch what did you want exactly, but i have fixed logic